Client/mods/deathmatch/logic/luadefs/CLuaWorldDefs.cpp

```diff
@@ -242,18 +252,38 @@ int CLuaWorldDefs::ProcessLineOfSight(lua_State* luaVM)
     argStream.ReadBool(flags.bSeeThroughStuff, false);
     argStream.ReadBool(flags.bIgnoreSomeObjectsForCamera, false);
     argStream.ReadBool(flags.bShootThroughStuff, false);
-    argStream.ReadUserData(pIgnoredElement, NULL);
+
+    if (argStream.NextIsTable()) // Is the next value a table? Read it as a user data table (will error if table contains invalid type)
+    {
+        argStream.ReadUserDataTable(vecIgnoredElements);
+    }
+    else if (argStream.NextIsNil()) // Is the next value nil? We can just skip to reading bIncludeBuildingInfo
+    {
+        argStream.Skip(1);
+    }
+    else if (!argStream.NextIsNone()) // Has the next value been supplied AT ALL? Then we'll try to read that argument as a userdata (which will ensure errors are returned upon incorrect type)
+    {
+        CClientEntity* pElement;
+        argStream.ReadUserData(pElement);
+        vecIgnoredElements.push_back(pElement);
+        while(!argStream.NextIsNone() && !argStream.NextIsBool()) // Keep trying to read each successive argument as userdata until the next argument has not been supplied, or it is a bool (bIncludeBuildingInfo).
+        {
+            CClientEntity* pElement;
+            argStream.ReadUserData(pElement);
+            vecIgnoredElements.push_back(pElement);
+        }
```

Comment:
I think you could `reserve(4096)` here. It might sound a lot, but this vector lives a very short life, and reserving a lot saves a lot of CPU.

Reply:
I will look into this, thanks.

Reply:
Any update?

Reply:
I think a table should be used here instead. Realistically people will have to pack it into a table anyways, and then they have to call unpack, so might as well accept a table instead.
I really have a bad mouth taste coming from `dxDrawPrimitive`.

Reply:
Not using vararg anymore, ReadUserDataTable reads into vector so this is resolved